src/main/java/net/sydokiddo/auditory/mixin/items/WeaponSwingMixin.java

```diff
@@ -25,110 +22,93 @@
+    private void auditory_playAxeSounds() {
+        level.playSound(null, this.getX(), this.getY(), this.getZ(), ModSoundEvents.ITEM_AXE_SWING, SoundSource.PLAYERS, 0.1F, 0.8f + this.level.random.nextFloat() * 0.4F);
     }
 
     @Unique
-    private static void playToolSounds(Entity entity) {
-        entity.level.playSound(null, entity.getX(), entity.getY(), entity.getZ(), ModSoundEvents.ITEM_TOOL_SWING, SoundSource.PLAYERS, 0.1F, 0.8f + entity.level.random.nextFloat() * 0.4F);
+    private void auditory_playToolSounds() {
+        level.playSound(null, this.getX(), this.getY(), this.getZ(), ModSoundEvents.ITEM_TOOL_SWING, SoundSource.PLAYERS, 0.1F, 0.8f + this.level.random.nextFloat() * 0.4F);
     }
 
     @Unique
-    private static void playLightweightToolSounds(Entity entity) {
-        entity.level.playSound(null, entity.getX(), entity.getY(), entity.getZ(), ModSoundEvents.ITEM_TOOL_SWING, SoundSource.PLAYERS, 0.1F, 1.4f + entity.level.random.nextFloat() * 0.8F);
+    private void auditory_playLightweightToolSounds() {
+        level.playSound(null, this.getX(), this.getY(), this.getZ(), ModSoundEvents.ITEM_TOOL_SWING, SoundSource.PLAYERS, 0.1F, 1.4f + this.level.random.nextFloat() * 0.8F);
     }
 
     @Unique
-    private static void playEmptyHandSounds(Entity entity) {
-        entity.level.playSound(null, entity.getX(), entity.getY(), entity.getZ(), SoundEvents.PLAYER_ATTACK_WEAK, SoundSource.PLAYERS, 0.1F, 0.8f + entity.level.random.nextFloat() * 0.4F);
+    private void auditory_playEmptyHandSounds() {
+        level.playSound(null, this.getX(), this.getY(), this.getZ(), SoundEvents.PLAYER_ATTACK_WEAK, SoundSource.PLAYERS, 0.1F, 0.8f + this.level.random.nextFloat() * 0.4F);
     }
 
-    @Inject(at = @At("HEAD"), method = "swing(Lnet/minecraft/world/InteractionHand;Z)V")
-    public void swing(InteractionHand interactionHand, boolean bl, CallbackInfo ci) {
-        if (!this.swinging || this.swingTime >= this.getCurrentSwingDuration() / 2 || this.swingTime < 0) {
-            this.swingTime = -1;
-            this.swinging = true;
-            this.swingingArm = interactionHand;
-            ItemStack itemStack = this.getItemInHand(interactionHand);
-            if (this.level instanceof ServerLevel) {
-                ClientboundAnimatePacket clientboundAnimatePacket = new ClientboundAnimatePacket(this, interactionHand == InteractionHand.MAIN_HAND ? 0 : 3);
-                ServerChunkCache serverChunkCache = ((ServerLevel) this.level).getChunkSource();
+    @Inject(
+        at = @At(value = "NEW",
+            target = "net/minecraft/network/protocol/game/ClientboundAnimatePacket"),
+        method = "swing(Lnet/minecraft/world/InteractionHand;Z)V"
+    )
+    private void auditory_swingSounds(InteractionHand interactionHand, boolean bl, CallbackInfo ci) {
+        ItemStack itemStack = this.getItemInHand(interactionHand);
 
-                // Sword Sounds
+        // TODO Fabric tool tags support
```

Comment:
Fabric has tool tags that allow items to have multiple tool types. I'm not sure how this should be handled, so I've left a note for the future.